pkg/metrics/metrics.go

```diff
@@ -17,12 +17,62 @@ limitations under the License.
 package metrics
 
 import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
 	"github.com/prometheus/client_golang/prometheus"
 	dto "github.com/prometheus/client_model/go"
 
 	"k8s.io/kube-state-metrics/pkg/options"
 )
 
+// Metric represents a single line entry in the /metrics export format
+type Metric string
+
+// NewMetric returns a new Metric
+func NewMetric(name string, labelKeys []string, labelValues []string, value float64) (*Metric, error) {
```

Comment:
I'm not sure how performance critical this end is as it just fills the cache, but the size of the buffer to write these bytes into should be calculatable upfront and then all bytes just directly written into it. But I think on this one we should first figure out whether this is actually a problem. To do that we should add some metrics on the metrics generation.